Sources/Sentry/SentryMigrateSessionInit.m

```diff
@@ -0,0 +1,117 @@
+#import <Foundation/Foundation.h>
+
+#import "SentryEnvelope.h"
+#import "SentryEnvelopeItemType.h"
+#import "SentryLog.h"
+#import "SentryMigrateSessionInit.h"
+#import "SentrySerialization.h"
+#import "SentrySession+Private.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@implementation SentryMigrateSessionInit
+
++ (void)migrateSessionInit:(NSString *)envelopeFilePath
+          envelopesDirPath:(NSString *)envelopesDirPath
+         envelopeFilePaths:(NSArray<NSString *> *)envelopeFilePaths;
+{
+    NSData *envelopeData = [[NSFileManager defaultManager] contentsAtPath:envelopeFilePath];
+    SentryEnvelope *envelope = [SentrySerialization envelopeWithData:envelopeData];
+    if (nil == envelope) {
+        return;
+    }
+
+    for (SentryEnvelopeItem *item in envelope.items) {
+        if ([item.header.type isEqualToString:SentryEnvelopeItemTypeSession]) {
+            SentrySession *session = [SentrySerialization sessionWithData:item.data];
+            if (nil != session && [session.flagInit boolValue]) {
+                [self setInitFlagOnNextEnvelopWithSameSessionId:session
+                                               envelopesDirPath:envelopesDirPath
+                                              envelopeFilePaths:envelopeFilePaths];
+            }
+        }
+    }
+}
+
++ (void)setInitFlagOnNextEnvelopWithSameSessionId:(SentrySession *)session
+                                 envelopesDirPath:(NSString *)envelopesDirPath
+                                envelopeFilePaths:(NSArray<NSString *> *)envelopeFilePaths
+{
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+
+    for (NSString *envelopeFilePath in envelopeFilePaths) {
+        NSString *envelopePath = [envelopesDirPath stringByAppendingPathComponent:envelopeFilePath];
+        NSData *envelopeData = [fileManager contentsAtPath:envelopePath];
+        SentryEnvelope *envelope = [SentrySerialization envelopeWithData:envelopeData];
```

Comment:
If we fail to load the file (say it was removed from the file system [by another process] by the time we get this far). Would this just return `nil`?

Reply:
Good point. Yes, it would. I added a test for it and we also check for `nil` here now.